src/zmtp_engine.cpp

```diff
@@ -138,11 +137,6 @@ bool zmq::zmtp_engine_t::handshake ()
     if (_outsize == 0)
         set_pollout ();
 
-    if (_has_handshake_timer) {
```

Comment:
This is partially correct, for ZMTP version below 3 there is no mechanism_ready. You can follow the semantic of the following PR: https://github.com/zeromq/libzmq/pull/3913/files.

When engine_ready is called the timer should be canceled as well.

Reply:
Something I've been wondering is how to test with older versions of ZMTP?  Any suggestions for old versions of libzmq with which to test?  

I'm guessing that it would be pretty much impossible to test this as part of automated unit tests.

Reply:
You can test using the Stream socket or OS file descriptor, some examples here:
https://github.com/zeromq/libzmq/blob/master/tests/test_mock_pub_sub.cpp
https://github.com/zeromq/netmq/blob/master/src/NetMQ.Tests/ZMTPTests.cs#L38

Reply:
Right -- the updated test does exactly that to test connecting to non-0mq.  

The question is more about connecting to older versions of ZMTP -- afaict there is no way to force 0mq to present an older version of ZMTP to peers.